datajunction-query/djqs/utils.py

```diff
@@ -56,11 +57,18 @@ def get_metadata_engine() -> Engine:
     return engine
 
 
-def get_session() -> Iterator[Session]:
+def get_session(request: Request = None) -> Iterator[Session]:
     """
     Per-request session.
     """
+    schema = request.headers.get("tenant")
     engine = get_metadata_engine()
+    settings = get_settings()
+
+    if schema:
+        engine = engine.execution_options(schema_translate_map={None: schema})
+
+    settings.customSchema = request.headers.get("new_tenant")
```

Comment:
@sadath-12 can you help me understand why you need to modify the settings instance directly like this? Also, since you're pulling schema from the "tenant" header, you don't need this settings property at all anymore, right?

Reply:
Hi @samredai  , this is to tell dj to switch to the schema when we hit register schema endpoint (created internally) so dj can run migrations on it on runtime or else it will have to be restarted

Reply:
Thanks @sadath-12 I understand the intent here more. I don't think modifying the application settings is the right approach though. Instead we should parameterize the alembic migration scripts (I think I recall @shangyian proposing this). Then in your custom endpoint logic, you can programmatically set that schema parameter using the context.